ml/src/main/scala/org/apache/spark/ml/spark/models/svm/SVMModel.scala

```diff
@@ -47,10 +53,19 @@ class SVMModel private[svm](val selfKey: Key[SVMModel],
         throw H2O.unimpl
     }
 
+  private val meanImputation: Boolean = MissingValuesHandling.MeanImputation.equals(_parms._missing_values_handling)
+
   protected def score0(data: Array[Double], preds: Array[Double]): Array[Double] = {
     java.util.Arrays.fill(preds, 0)
+
     val pred =
-      data.zip(_output.weights).foldRight(_output.interceptor){ case ((d, w), acc) => d * w + acc}
+      data.zip(_output.weights).foldRight(_output.interceptor){ case ((d, w), acc) =>
+        if(meanImputation && lang.Double.isNaN(d)) {
```

Comment:
Yes it will be a NaN in that case, I can do that or throw an exception. Skipping NaN values doesn't make much sense computation wise (the MissingValuesHandling.Skip during training is skipping the whole row if it has NaNs, not singular columns).